packages/frontend/src/hooks/useTransactionSummary.ts

```diff
@@ -0,0 +1,39 @@
+import { Tx, TxType } from "@carrot-kpi/react";
+import { useEffect, useState } from "react";
+import { useTranslation } from "react-i18next";
+import { useProvider } from "wagmi";
+import { getTransactionSummary } from "../utils/transactions";
+
+export const useTransactionSummary = (
+    tx: Tx<TxType>
+): { loading: boolean; summary: string } => {
+    const { t } = useTranslation();
+    const provider = useProvider();
+
+    const [loading, setLoading] = useState(false);
+    const [summary, setSummary] = useState("");
+
+    useEffect(() => {
+        let cancelled = false;
+        const getSummary = async () => {
+            if (!cancelled) setLoading(true);
+            try {
+                const summary = await getTransactionSummary(t, provider, tx);
```

Comment:
If I'm not mistaken, `getTransactionSummary` returns a promise only when the `TxType` is `TxType.ERC20_APPROVAL`. Could it cause any issues in the other cases, since we're always expecting a promise as a result here?